simulator/src/main/cook/sim/reporting/groups.clj

```diff
@@ -0,0 +1,83 @@
+(ns cook.sim.reporting.groups
+  (:require [clojure.pprint :as pprint]
+            [datomic.api :as d]
+            [cook.sim.database :as data]
+            [cook.sim.util :as u]))
+
+
+(defn find-groups
+  "Given a collection of job-results, returns only the jobs with groups,
+  in a map keyed by group name."
+  [job-results]
+  (dissoc (group-by :group job-results) nil))
+
+(defn time-of-first-request
+  "Given a collection of job results, return the earliest time that any was requested."
+  [job-results]
+  (->> job-results (map :requested-at) (apply min)))
+
+(defn sorted-start-times
+  "Returns the start times for jobs that HAVE start times, ordered"
+  [job-results]
+  (->> job-results (map :started-at) (filter identity) sort))
```

Comment:
`(remove nil?)` makes the intent clearer of removing the nil values instead of `(filter identity)`